protocol/messenger_communities.go

```diff
@@ -59,6 +63,56 @@ func (m *Messenger) publishOrgInvitation(org *communities.Community, invitation
 	return err
 }
 
+func (m *Messenger) handleCommunitiesHistoryArchivesSubscription(c chan *communities.Subscription) {
+
+	go func() {
+		for {
+			select {
+			case sub, more := <-c:
+				if !more {
+					return
+				}
+
+				if sub.CreatingHistoryArchivesSignal != nil {
+					m.config.messengerSignalsHandler.CreatingHistoryArchives(sub.CreatingHistoryArchivesSignal.CommunityID)
+				}
+
+				if sub.HistoryArchivesCreatedSignal != nil {
+					m.config.messengerSignalsHandler.HistoryArchivesCreated(
+						sub.HistoryArchivesCreatedSignal.CommunityID,
+						sub.HistoryArchivesCreatedSignal.From,
+						sub.HistoryArchivesCreatedSignal.To,
+					)
+				}
+
+				if sub.NoHistoryArchivesCreatedSignal != nil {
+					m.config.messengerSignalsHandler.NoHistoryArchivesCreated(
+						sub.NoHistoryArchivesCreatedSignal.CommunityID,
+						sub.NoHistoryArchivesCreatedSignal.From,
+						sub.NoHistoryArchivesCreatedSignal.To,
+					)
+				}
```

Comment:
These look a bit unfortunate. We emit the `History*Signal`'s on the `Subscription` and then use the attached data to send them to connected clients by destructing the attached data, which is then reconstructed again.

The reason I do this is because we already need the `History*Signal`s anyways to inform connected clients, so it made sense to reuse them here in the subscription.

Lemme know if this should be done differently.